### PR TITLE
oidc: document oidc prefix flags and add note about Okta

### PR DESCRIPTION
**Pull request by ericchiang**
Document the new `--oidc-[group|username]-prefix` flags and add a note about Okta's refreshing. Also remove the "extra-scopes" field since it's no longer used (https://github.com/kubernetes/kubernetes/pull/45317).

cc @kubernetes/sig-auth-pr-reviews 

Docs for https://github.com/kubernetes/kubernetes/pull/50875
Closes https://github.com/kubernetes/kubernetes/issues/36847

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5884)

### CODE REVIEW

File: docs/admin/authentication.md [hunk 1]
```diff
@@ -294,19 +294,26 @@ To enable the plugin, configure the following flags on the API server:
...
 | `--oidc-groups-claim` | JWT claim to use as the user's group. If the claim is present it must be an array of strings. | groups | No |
+| `--oidc-groups-prefix` | Prefix prepended to group claims to prevent clashes with existing names (such as `system:` groups). Setting the value `oidc:` would create group names like `oidc:engineering` and `oidc:infra`. | `oidc:` | No |
 | `--oidc-ca-file` | The path to the certificate for the CA that signed your identity provider's web certificate.  Defaults to the host's root CAs. | `/etc/kubernetes/ssl/kc-ca.pem` | No |
 
-If a claim other than `email` is chosen for `--oidc-username-claim`, the value
-will be prefixed with the `--oidc-issuer-url` to prevent clashes with existing
-Kubernetes names (such as the `system:` users). For example, if the provider
-URL is `https://accounts.google.com` and the username claim maps to `jane`, the
-plugin will authenticate the user as:
+If `--oidc-username-prefix` isn't provided and `--oidc-username-claim` is set to
```
Comment (liggitt) [inline docs/admin/authentication.md]: can we hoist this into the table? I expected to see the default behavior and opt out info by the param there

Comment (ericchiang) [reply docs/admin/authentication.md]: yeah, update

File: docs/admin/authentication.md [hunk 2]
```diff
@@ -337,7 +329,9 @@ Setup instructions for specific systems:
 
 ##### Option 1 - OIDC Authenticator
 
-The first option is to use the `oidc` authenticator.  This authenticator takes your `id_token`, `refresh_token` and your OIDC `client_secret` and will refresh your token automatically.  Once you have authenticated to your identity provider:
+The first option is to use the kubectl `oidc` authenticator, which set the `id_token` as a bearer token for all requests and refreshes the token once it expires. After you've logged into your provider, use kubectl to add your `id_token`, `refresh_token`, `client_id`, and `client_secret` to configure the plugin.
+
+Providers that don't return an `id_token` as part of their refresh token response (e.g. [Okta](https://developer.okta.com/docs/api/resources/oidc.html#response-parameters-4)) aren't supported by this plugin and should use "Option 2" below.
```
Comment (steveperry-53) [inline docs/admin/authentication.md]: which set -> which sets

**Comment by steveperry-53** [comment]
@ericchiang @liggitt One comment in the text. Otherwise, Docs LGTM.

File: docs/admin/authentication.md (hunk 2 above)
Comment (ericchiang) [reply docs/admin/authentication.md]: thanks for the review. updated

**Comment by steveperry-53** [comment]
@liggitt Are you ready to give tech approval? Thanks.

**Pull request merged**